filebeat/beater/filebeat.go

```diff
@@ -355,10 +353,9 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 
 	// Register reloadable list of inputs and modules
 	inputs := cfgfile.NewRunnerList(management.DebugK, inputLoader, fb.pipeline)
-	reload.Register.MustRegisterList("filebeat.inputs", inputs)
+	reload.RegisterV2.MustRegisterList("input", inputs)
```

Comment:
Can we put the V1 vs V2 protocol decision behind a feature flag (probably an undocumented configuration flag)? 

I think we want to be able to switch between the V1 and V2 implementations without requiring an emergency release to revert it in case of of problems the first time a user tries this.

Reply:
So, my understanding was that by the time this was made available to users, the V1 code could would be gone from the agent. Is there a plan to have the agent swap between v1/v2 now, or am I misunderstanding?

Reply:
The V1 co-exists as a separate RPC in the same proto file: https://github.com/elastic/elastic-agent-client/blob/b0328d2fe484f3ecfe6d52efdf18058431743e83/elastic-agent-client.proto#L342

The V1 code cannot be removed until every input (all beats, endpoint security, APM, etc) migrate to V2 so the agent has to support both.

On the client/input side I think it is as simple as switching which agent RPC we invoke to switch between the V1 and V2 protocols. The agent already supports both side by side. @ph or @blakerouse can correct me if this understanding is incorrect.

Reply:
8.5 release of Elastic Agent will only support the v2 protocol, v1 will not stay around.

Reply:
As a precaution it would be nice to have a way to revert to v1 in case of an unexpected issue after 8.5 is released. I am usually always in favour of feature flags for big changes.

If doing that is complex though I can buy into not supporting a fallback mechanism.